fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/fluid/FluidPreconditions.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transfer.v1.fluid;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.Fluids;
+
+/**
+ * Preconditions for fluid transfer.
+ */
+public class FluidPreconditions {
+	/**
+	 * Ensure that the passed fluid is not empty.
+	 *
+	 * @throws IllegalArgumentException If the fluid is empty.
+	 */
+	public static void notEmpty(Fluid fluid) {
+		if (fluid == Fluids.EMPTY) {
+			throw new IllegalArgumentException("Fluid may not be empty.");
+		}
+	}
```

Comment:
```suggestion
public static void validateInsert(Fluid fluid) {
    if(fluid == Fluids.EMPTY || !fluid.isStill()) {
      throw new IllegalArgumentException("Fluid may not be empty or flowing.");
   }
}
```

Reply:
It's used both for insert and extract - but yes, we should probably do something about flowing fluids? Putting them in such a precondition would ensure nobody ever uses them - it would just crash at any time. I'll think a bit more about it.

Reply:
`FluidKey`s don't accept flowing fluids.